Basic-Car-Maintenance/Shared/Settings/ViewModels/SettingsViewModel.swift

```diff
@@ -45,21 +49,32 @@ final class SettingsViewModel: ObservableObject {
         }
     }
     
-    func addVehicle(_ vehicle: Vehicle) async {
-        
+    /// Adds a new vehicle to the Firestore database and the local 'vehicles' array.
+    ///
+    /// - Parameter vehicle: The vehicle object to be added.
+    /// - Throws: An error if there's an issue adding the vehicle to Firestore.
+    ///
+    /// This function adds a new vehicle object to the Firestore database under the 'vehicles' collection.
```

Comment:
For all your comments I don't think this 2nd part is necessary, instead just have it all explained in the first part of the comment